notification-server/src/bin/service.rs

```diff
@@ -346,6 +355,36 @@ async fn traverse_chain(
     processed_height
 }
 
+async fn catch_up_to_limit(
```

Comment:
Would be nice with a docstring here, explaining what is returned and what this height can be used for.